server/src/main/java/org/apache/druid/server/security/AuthorizationUtils.java

```diff
@@ -263,6 +263,10 @@ public static Access authorizeAllResourceActions(
       throw new ISE("No authorizer found with name: [%s].", authenticationResult.getAuthorizerName());
     }
 
+    if (authorizer instanceof AllowAllAuthorizer) {
```

Comment:
earlier this code will filter a resource if `resourceActionGenerator.apply(resource)` returns null but now it won't. does that difference matter?

Reply:
Yeah, looks like this isn't the right thing to do. 
Also, the following loop needs to be executed because there could be duplicates in `Iterable<ResourceAction`. Will file an issue and a PR to revert this change.

Reply:
do you intend to get rid of the duplicates or keep them? From the description in https://github.com/apache/druid/issues/11012, it seems we want to de-dup the entries though that wasn't happening earlier.